project_purchase_line_link/models/project_project.py

```diff
@@ -0,0 +1,34 @@
+# © 2025 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import models, _, fields, api
+
+
+class ProjectProject(models.Model):
+    _inherit = "project.project"
+
+    purchase_order_lines_count = fields.Integer(compute='_compute_purchase_order_lines_count', groups='purchase.group_purchase_user')
+
+    @api.depends('analytic_account_id')
+    def _compute_purchase_order_lines_count(self):
+        purchase_orders_line = self.env['purchase.order.line'].search([
+            ('account_analytic_id', '=', self.analytic_account_id.ids)
+        ])
+        for project in self:
+            project.purchase_order_lines_count = len(purchase_orders_line)
+
+    def action_open_purchase(self):
+        purchase_orders_line = self.env['purchase.order.line'].search([
+            ('account_analytic_id', 'in', self.analytic_account_id.ids)
+        ])
+        action_window = {
+            'name': _('Purchase Order Lines'),
+            'type': 'ir.actions.act_window',
+            'res_model': 'purchase.order.line',
+            'views': [[False, 'tree'], [False, 'form']],
+            'domain': [('id', 'in', purchase_orders_line.ids)],

```

Comment:
Should this simpler code work? Not tested...
```suggestion
            'domain': [('account_analytic_id', '=', self.analytic_account_id.id)],

```
If it works, two more adjustments should be needed:

* Prefix code with `self.ensure_one()`, because such action only will fired with one record... and we actually need it
* Remove code below for view selection (a lot of PO lines will be expected for a project, only 1 line will be very strage case)